### ISSUE TITLE
Custom message for UniqueConstraintError

### ISSUE THREAD

**Issue by MichaelRo** [sequelize/sequelize#7072]
Hello dear sequelizers,

**TL;DR: The custom UniqueConstraintError message I've set through the 'msg' attribute doesn't show up, I think it has something to do with the name of the unique key.**

## What you are doing?
I'm trying to update sequelize@3.4.1 to version 3.28.0.
One of the problems I've encountered was the custom message of the UniqueConstraintError.

Here's my model:
```js
var Entity = sequelize.define("Entity", {
        id: {
            type: DataTypes.INTEGER.UNSIGNED,
            allowNull: false,
            primaryKey: true,
            autoIncrement: true,
            validate: {
                isInt: {
                    msg: errorMessages.isInt()
                }
            }
        },
        name: {
            type: DataTypes.STRING(128),
            allowNull: false,
            unique: {
                name: "uniqueNamePerAccount",
                args: [["Name", "Account ID"]],
                msg: errorMessages.unique()
            },
            validate: {
                notEmpty: {
                    msg: errorMessages.notEmpty()
                }
            }
        },
        accountId: {
            type: DataTypes.INTEGER.UNSIGNED,
            allowNull: false,
            unique: {
                name: "uniqueNamePerAccount",
                args: [["Name", "Account ID"]],
                msg: errorMessages.unique()
            },
            validate: {
                isInt: {
                    msg: errorMessages.isInt()
                }
            }
        }
}
```

Here's my custom message object:
```js
function unique() {
    return {
        innercode: innerErrorCodes.unique,
        data: {},
        message: util.format("There is already an item that contains the exact same values of the following keys")
    };
}
```

## What do you expect to happen?
I expected the message to be part of the error object, but It's just doesn't show up.
I tried to change my custom message to a string but it didn't help.

## What is actually happening?
So actually I tried to debug sequelize, and I came across query.js (mysql version).
Here's the code (lines 122-154):
```js
Query.prototype.formatError = function (err) {
  var match;

  switch (err.errno || err.code) {
    case 1062:
      match = err.message.match(/Duplicate entry '(.*)' for key '?((.|\s)*?)'?$/);

      var values = match ? match[1].split('-') : undefined
        , fields = {}
        , message = 'Validation error'
        , uniqueKey = this.model && this.model.uniqueKeys[match[2]];

      if (!!uniqueKey) {
        if (!!uniqueKey.msg) message = uniqueKey.msg;
        fields = Utils._.zipObject(uniqueKey.fields, values);
      } else {
        fields[match[2]] = match[1];
      }

      var errors = [];
      var self = this;
      Utils._.forOwn(fields, function(value, field) {
        errors.push(new sequelizeErrors.ValidationErrorItem(
          self.getUniqueConstraintErrorMessage(field),
          'unique violation', field, value));
      });

      return new sequelizeErrors.UniqueConstraintError({
        message: message,
        errors: errors,
        parent: err,
        fields: fields
      });
```

These are the values in debug:
`err.message = "ER_DUP_ENTRY: Duplicate entry '49991365' for key 'PRIMARY'"`
`match = ["Duplicate entry '49991365' for key 'PRIMARY'",
                  "49991365",
                  "PRIMARY",
                  "Y"]`
`match[2] = "PRIMARY"`
`uniqueKey = undefined`

So instead of taking my custom message, the code took no message.

It seems like it has something to do with the unique key, I think there's a problem with the if clause.

Thank you very much!

__Dialect:__ mysql
__Database version:__ 5.7
__Sequelize version:__ 3.28.0

**Comment by sushantdhiman** [comment]
Have you tried `master` branch, I recently did some work around validations in `master`
